python/semantic_kernel/connectors/memory/azure_ai_search/azure_ai_search_collection.py

```diff
@@ -91,7 +96,7 @@ def __init__(
             if not collection_name:
                 collection_name = search_client._index_name
             elif search_client._index_name != collection_name:
-                raise MemoryConnectorInitializationError(
+                raise VectorStoreInitializationException(
```

Comment:
Question: Is this a breaking change? If someone was catching this specific exception before, it's now all of a sudden not catching it.

Reply:
Unfortunately this does mean breaking changes, I've updated the description of this PR.